commands/addAuthor.ts

```diff
@@ -0,0 +1,140 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, MessageActionRow, MessageButton, MessageEmbed } from 'discord.js';
+import { AuthorInfo } from '../types/author';
+import { createAuthor, isAirtableError } from '../utils/airTableCalls';
+import HandledError, { isHandledError } from '../utils/error';
+import isValidUrl from '../utils/urlChecker';
+
+export const data = new SlashCommandBuilder()
+  .setName('add-author')
+  .setDescription('Adds an author to the knowledgebase')
+  .addStringOption(
+    option => option.setRequired(true)
+      .setName('author')
+      .setDescription('Enter the author full name'),
+  )
+  .addBooleanOption(option =>
+    option.setRequired(true)
+      .setName('is_dao_member')
+      .setDescription('Are they a Developer DAO member?'),
+  )
+  .addStringOption(
+    option => option.setRequired(false)
+      .setName('twitter_url')
+      .setDescription('Enter their Twitter URL'),
+  )
+  .addStringOption(
+    option => option.setRequired(false)
+      .setName('youtube_url')
+      .setDescription('Enter their YouTube URL'),
+  );
+
+
+function getSanitizedAuthorInfo(interaction: CommandInteraction): AuthorInfo {
+  const authorName = interaction.options.getString('author');
+  const isDaoMember = interaction.options.getBoolean('is_dao_member');
+  const twitterUrl = interaction.options.getString('twitter_url');
+  const youtubeUrl = interaction.options.getString('youtube_url');
+
+  if (authorName === null || isDaoMember === null) {
+    // The discord bot *should* handle this, but just in case.
+    throw new HandledError('Missing required fields');
+  }
+
+  return {
+    name: authorName,
+    isDaoMember,
+    twitterUrl: twitterUrl || '',
+    youtubeUrl: youtubeUrl || '',
+  }
+}
+
+export async function execute(interaction: CommandInteraction) {
+  const REPLY = {
+    YES: 'yes',
+    NO: 'no',
+  };
+
+  const author = getSanitizedAuthorInfo(interaction);
+  const { name, isDaoMember, twitterUrl, youtubeUrl } = author;
+  if (twitterUrl !== '' && !isValidUrl(twitterUrl)) {
+    await interaction.reply({
+      content: 'Invalid Twitter URL: ${twitterUrl}',
+      ephemeral: true,
+    })
+    return;
+  }
```

Comment:
I had a quick look and it didn't seem straightforward.

The first challenge is about the messages being "ephemeral". Because they are supposed to be seen only by the user that triggered the command, there would have to be a way for the user to also reply in the same "ephemeral" way - i.e.: only seen by the bot but not by other people.

Reply:
This ability to change/correct things is one of the reasons I was looking at using a DM Channel for adding a resource, but having as much of the process run in these interactions and using ephemeral messages feels better.